python/ray/_private/gc_collect_manager.py

```diff
@@ -0,0 +1,64 @@
+import gc
+import logging
+import threading
+import time
+from typing import Callable, Optional
+
+logger = logging.getLogger(__name__)
+
+
+class PythonGCThread(threading.Thread):
+    """A background thread that triggers Python garbage collection.
+    This thread waits for GC events from CoreWorker and triggers `gc.collect()` when
+    requested, ensuring that collections are spaced out by at least
+    `min_interval` seconds."""
+
+    def __init__(
+        self, min_interval: int = 5, gc_collect_func: Optional[Callable] = None
+    ):
+        logger.debug("Starting Python GC thread")
+        super().__init__(name="PythonGCThread", daemon=True)
+        self._running = True
+        self._last_gc_time = float("-inf")
+        self._min_gc_interval = min_interval
+        self._gc_event = threading.Event()
+        # Set the gc_collect_func for UT, defaulting to gc.collect if None
+        self._gc_collect_func = gc_collect_func or gc.collect
+
+    def trigger_gc(self) -> None:
+        self._gc_event.set()
+
+    def run(self):
+        while self._running:
+            self._gc_event.wait()
+            self._gc_event.clear()
+
+            if not self._running:
+                break
+
+            time_since_last_gc = time.perf_counter() - self._last_gc_time
```

Comment:
no need to use `perf_counter` here -- it's more expensive since it's high resolution